### PR TITLE
Add electronically filed CS01 form

### PR DESCRIPTION
**Pull request by danfalc**
- Create new SQL file for company that has CS01 due
- Create XML document for the electronic CS01
- Create new package for XML documents
- Move SQL files to separate package
- XmlHelper class to replace the placeholders in the XML files
- Update Fes Processor to deal with change to env.conf chips-rest-api
- Add method in DbUtil to get confirmation statement date for current company
- Add wait to TeamWorkPage to avoid flakiness

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/data/dbutil/sql/CompanySql.java [hunk 1]
```diff
@@ -40,7 +42,7 @@ private void loadFromFile(String resourceIn) {
             String resource = resourceIn.replaceAll("^/+", "");
             BufferedReader reader = new BufferedReader(
                     new InputStreamReader(
-                            Objects.requireNonNull(getClass().getResourceAsStream("/sql/" + resource))));
+                            Objects.requireNonNull(getClass().getResourceAsStream("/uk/gov/companieshouse/sql/" + resource))));
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/data/dbutil/sql/CompanySql.java:45]: Does it need the full path? Were you having issues here?

File: src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import uk.gov.companieshouse.data.dbutil.DbUtil;
+import uk.gov.companieshouse.data.dbutil.sql.CompanySql;
+import uk.gov.companieshouse.testdata.DocumentDetails;
+import uk.gov.companieshouse.utils.BarcodeGenerator;
+import uk.gov.companieshouse.utils.FesProcessor;
+import uk.gov.companieshouse.utils.TestContext;
+import uk.gov.companieshouse.utils.XmlHelper;
+
+
+public class ElectronicFilingStepDefs {
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java:20]: Why `ElectronicFilingStepDefs` and not `ConfirmationStatementStepDefs`?

File: src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+    /**
+     * Simulate a submission of a CS01 form that has been filed through electronic filing.
+     */
+    @When("I process a no update e-filed CS01 form for a private limited company")
+    public void processElectronicFiledCS01Form() throws IOException {
+        String filename = "test.xml";
+
+        SimpleDateFormat xmlDateFormatter = new SimpleDateFormat("yyyy-MM-dd");
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java:51]: Haven't we got common code to provide today's date?
Or is this one format specific?

File: src/test/resources/uk/gov/companieshouse/features/cs01.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
+@confirmation_statement @cs01
+Feature: CS01 - Confirmation Statement.
+
+* This form is the replacement for the old Annual Return.
```
Comment (rsuller) [inline src/test/resources/uk/gov/companieshouse/features/cs01.feature:4]: Fair play this really does cover everything you could ever want to know about a CS01 😂

**Review by rsuller — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java (hunk 3 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java:51]: I've used the common one below. This is format specific to XML. Maybe worth adding this to the common class as more XML files are added

File: src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java (hunk 2 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/stepdefinitions/ElectronicFilingStepDefs.java:20]: Will be used to deal with other forms that are electronically filed similar to FesProcessingStepDefs

File: src/test/java/uk/gov/companieshouse/data/dbutil/sql/CompanySql.java (hunk 1 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/data/dbutil/sql/CompanySql.java:45]: Yep this is due to the addition of the new directory

File: conf/env.conf [hunk 5]
```diff
@@ -1,11 +1,12 @@
 testing {
-chips_url=
-username=
-password=
-chips-jdbc=
-chips-db-user=
-chips-db-pass=
-chips-rest-api-uri =
+chips_url="https://chips-staging.companieshouse.gov.uk/chips/cff"
```
[the remark this answers is not in the document]
Comment (rsuller) [reply conf/env.conf]: It shouldn't be tracking changes in this file now. So you may need to clear the cache. This should be added to '.gitignore'

[the remark this answers is not in the document]
Comment (danfalc) [reply conf/env.conf]: Sorted now anyway. Removed everything that shouldn't be there

**Pull request merged**